### ISSUE TITLE
Check algorithm parameters for validity

### ISSUE THREAD

**Issue by jkbradley** [graphframes/graphframes#76]
For the stdlib algorithms (PageRank, etc.), we should check for valid parameter values in the setters, rather than failing somewhere internally.

**Comment by bcajes** [comment]
Hi, in the process of familiarizing myself with this library, I've decided give this one a shot.  Let me know if the following is more or less what is needed:  
https://github.com/graphframes/graphframes/compare/master...bcajes:master 
If so, I'll add checks for the remaining algorithms.

**Comment by jkbradley** [comment]
Yep, adding checks in the setter methods is right.  Just one comment about tests: Can you please use this pattern which we use in Spark itself?
```
withClue("explanation...") {
  intercept[MyExceptionType] {
    // code which causes exception
  }
}
```
Thanks!
